### ISSUE TITLE
failed to enable bitrot

### ISSUE THREAD

**Issue by Madhu-1** [gluster/glusterd2#815]
**step to reproduce**

1. createvolume 
2. start volume
3. enable bitrot on volume
```
./build/glustercli  bitrot enable test_volume
Failed to enable bitrot for volume test_volume


Error: Request failed with HTTP Status code 500
Step bitrot-enable.Commit failed on 2 nodes

```

4.try to enable bitrot again

```
./build/glustercli --glusterd-host=10.70.43.235 bitrot enable test_volume
Failed to enable bitrot for volume test_volume


Error: Request failed with HTTP Status code 400
Bitrot is already enabled

```
**Expected**: Error response should be same in above both conditions

5. try to get vol info, volume info shows bitrot option is enabled 
**bitrot option shouldn't be enabled** 
```
 [
    {
        "id": "335b6682-4b1f-429a-96c9-429777dbe10d",
        "name": "test_volume2",
        "type": "Distribute",
        "transport": "tcp",
        "distribute-count": 4,
        "replica-count": 0,
        "arbiter-count": 0,
        "options": {
            "bit-rot.scrubber": "true",
            "bitrot-stub.bitrot": "on"
        },
        "state": "Started",
        "subvols": [
            {
                "name": "test_volume2-distribute-0",
                "type": "Distribute",
                "bricks": [
                    {
                        "id": "a3b931a5-bd58-4e9f-8247-d8d6c3d3f927",
                        "path": "/export/brick1/data",
                        "volume-id": "335b6682-4b1f-429a-96c9-429777dbe10d",
                        "volume-name": "test_volume2",
                        "peer-id": "240f4082-e951-4f52-85ff-46210efd4b78",
                        "host": "dhcp42-235.lab.eng.blr.redhat.com",
                        "type": "Brick"
                    }
                ],
                "replica-count": 1,
                "arbiter-count": 0
            },
            {
                "name": "test_volume2-distribute-1",
                "type": "Distribute",
                "bricks": [
                    {
                        "id": "872a55de-bf5b-4bc7-98ca-7306483f5e0a",
                        "path": "/export/brick2/data",
                        "volume-id": "335b6682-4b1f-429a-96c9-429777dbe10d",
                        "volume-name": "test_volume2",
                        "peer-id": "9f62c6db-944e-4e5c-a5c0-658e4fea369b",
                        "host": "dhcp43-75.lab.eng.blr.redhat.com",
                        "type": "Brick"
                    }
                ],
                "replica-count": 1,
                "arbiter-count": 0
            },
            {
                "name": "test_volume2-distribute-2",
                "type": "Distribute",
                "bricks": [
                    {
                        "id": "f921e1b3-06f5-4e2c-b15a-002bd4b0ee68",
                        "path": "/export/brick3/data",
                        "volume-id": "335b6682-4b1f-429a-96c9-429777dbe10d",
                        "volume-name": "test_volume2",
                        "peer-id": "240f4082-e951-4f52-85ff-46210efd4b78",
                        "host": "dhcp42-235.lab.eng.blr.redhat.com",
                        "type": "Brick"
                    }
                ],
                "replica-count": 1,
                "arbiter-count": 0
            },
            {
                "name": "test_volume2-distribute-3",
                "type": "Distribute",
                "bricks": [
                    {
                        "id": "a3c6f351-f06d-48c7-a076-fe45e95400ed",
                        "path": "/export/brick4/data",
                        "volume-id": "335b6682-4b1f-429a-96c9-429777dbe10d",
                        "volume-name": "test_volume2",
                        "peer-id": "9f62c6db-944e-4e5c-a5c0-658e4fea369b",
                        "host": "dhcp43-75.lab.eng.blr.redhat.com",
                        "type": "Brick"
                    }
                ],
                "replica-count": 1,
                "arbiter-count": 0
            }
        ],
        "metadata": {},
        "snap-list": null
    }
]
```
6. volfile info

```
volume test_volume2-distribute-0
    type cluster/distribute
end-volume

volume test_volume2-distribute-1-client-0
    type protocol/client
    option remote-host dhcp43-75.lab.eng.blr.redhat.com
    option remote-subvolume /export/brick2/data
end-volume

volume test_volume2-distribute-1
    type cluster/distribute
    subvolumes test_volume2-distribute-1-client-0
end-volume

volume test_volume2-distribute-2
    type cluster/distribute
end-volume

volume test_volume2-distribute-3-client-0
    type protocol/client
    option remote-host dhcp43-75.lab.eng.blr.redhat.com
    option remote-subvolume /export/brick4/data
end-volume

volume test_volume2-distribute-3
    type cluster/distribute
    subvolumes test_volume2-distribute-3-client-0
end-volume

volume test_volume2-bit-rot-0
    type features/bit-rot
    subvolumes test_volume2-distribute-0 test_volume2-distribute-1 test_volume2-distribute-2 test_volume2-distribute-3
end-volume

volume bitd
    type debug/io-stats
    subvolumes test_volume2-bit-rot-0
end-volume

```
7.  Logs for failed bitrot enable

```
INFO[2018-05-18 12:45:33.673073] client connected                              address="10.70.43.235:49147" server=sunrpc source="[server.go:158:sunrpc.(*SunRPC).acceptLoop]" transport=tcp
ERRO[2018-05-18 12:45:33.687713] volume not found                              source="[store-utils.go:62:volume.GetVolume]" volume=01bb9221-4961-4fd8-a8bd-a92b7a475eac-gluster/bitd
ERRO[2018-05-18 12:45:33.687737] failed to get volinfo from store              error="volume not found" source="[handshake_prog.go:127:sunrpc.(*GfHandshake).ServerGetspec]" volume=01bb9221-4961-4fd8-a8bd-a92b7a475eac-gluster/bitd
DEBU[2018-05-18 12:45:33.689067] Child exited                                  pid=12135 source="[daemon.go:87:daemon.Start]" status="exit status 1"
ERRO[2018-05-18 12:45:33.689106] Step failed on node.                          error="exit status 1" node=01bb9221-4961-4fd8-a8bd-a92b7a475eac reqid=65ea8b63-1505-4d67-b024-6028b1425376 source="[step.go:74:transaction.runStepFuncOnNodes]" step=bitrot-enable.Commit txnid=9f262b2e-5d35-4fa4-a61b-2be19b429ad8
DEBU[2018-05-18 12:45:33.689329] Signal received                               signal="child exited" source="[main.go:135:main.main]"
INFO[2018-05-18 12:45:33.689420] client disconnected                           address="10.70.43.235:49147" server=sunrpc source="[server.go:114:sunrpc.(*SunRPC).pruneConn]"
ERRO[2018-05-18 12:45:33.829893] TxnSvc.Runstep failed on peer                 error="exit status 1" remotepeer="291e7f91-170b-4ed6-887e-43f98c7c319b(dhcp43-75.lab.eng.blr.redhat.com)" reqid=65ea8b63-1505-4d67-b024-6028b1425376 source="[rpc-client.go:78:transaction.runStepOn]" txnid=9f262b2e-5d35-4fa4-a61b-2be19b429ad8
ERRO[2018-05-18 12:45:33.829985] Step failed on node.                          error="exit status 1" node=291e7f91-170b-4ed6-887e-43f98c7c319b reqid=65ea8b63-1505-4d67-b024-6028b1425376 source="[step.go:74:transaction.runStepFuncOnNodes]" step=bitrot-enable.Commit txnid=9f262b2e-5d35-4fa4-a61b-2be19b429ad8
ERRO[2018-05-18 12:45:33.830007] Transaction failed, rolling back changes      error="Step bitrot-enable.Commit failed on 2 nodes" reqid=65ea8b63-1505-4d67-b024-6028b1425376 source="[transaction.go:128:transaction.(*Txn).Do]" txnid=9f262b2e-5d35-4fa4-a61b-2be19b429ad8
ERRO[2018-05-18 12:45:33.830022] failed to enable bitrot                       error="Step bitrot-enable.Commit failed on 2 nodes" reqid=65ea8b63-1505-4d67-b024-6028b1425376 source="[rest.go:101:bitrot.bitrotEnableHandler]" volname=test_volume

```

**Comment by rishubhjain** [comment]
@Madhu-1 In addition to this, even after the failure the bitrot is marked as enabled for that volume.

**Comment by kotreshhr** [comment]
The log says volume not found error. Was the volume name given different?

**Comment by rishubhjain** [comment]
@kotreshhr Referring to the etcd data provided in the 5th step, the volume name provided is the same. May be @Madhu-1 can verify it.

**Comment by kotreshhr** [comment]
Is it consistently reproducible ? If volume name is different and we are setting the enable flag, that also needs a fix. Let me know the exact scenario.

**Comment by Madhu-1** [comment]
@kotreshhr  yes am able to reproduce it 5/5.

**step to reproduce the issue**

1.createvolume
2.start volume
3.enable bitrot on volume

**Comment by Madhu-1** [comment]
@aravindavk  updated volfile info and volume info

**Comment by rishubhjain** [comment]
@Madhu-1 This issue can be closed now I guess

**Comment by Madhu-1** [comment]
fixed by #824 and #827
